docs/docs/architecture/client_broker_protocol.md

```diff
@@ -0,0 +1,1223 @@
+
+As mentioned before, opening a queue is a two-step process.  The first one
+`OpenQueue`/`OpenQueueResponse`.  Let's look at the second step.  Note that a
+BlazingMQ client cannot start producing/consumer to/from a queue until the
+second step is complete.
+
+#### `ConfigureStream` Request
+
+As the name suggests, the
+[`ConfigureStream`](https://github.com/bloomberg/blazingmq/blob/69f0f8f5b188ca1eb07b9d262093f949729db538/src/groups/bmq/bmqp/bmqp_ctrlmsg.xsd#L274)
+request provides a way to the BlazingMQ client to configure its "connection"
+with the queue.  Note that the connection is being used logically here, and
+should not be confused with a TCP connection.  This request is mostly a no-op
+for producers in the BlazingMQ backend (but the producers still need to send
+it).
+
+This request is very important for consumers in many ways:
+
+- It enables a consumer to specify one or more subscriptions.
+
+- It enables a consumer to dynamically alter its subscriptions at any time.
+
+- It enables a consumer to configure various attributes of a
+  [subscription](../../features/subscriptions), like priority, [flow
+  control](../../features/consumer_flow_control), etc.
+
+Now that we have a high level understanding of this request, let's dig deeper.
+
+The `ConfigureStream` request contains two fields.  The first one is `qId`,
+which was explained above.  The second one is
+[`streamParameters`](https://github.com/bloomberg/blazingmq/blob/9b692fe25f74543e954a27e30b6b15b0ae057c8d/src/groups/bmq/bmqp/bmqp_ctrlmsg.xsd#L287C22-L287C38).
+Let's look at various fields of this
+[type](https://github.com/bloomberg/blazingmq/blob/9b692fe25f74543e954a27e30b6b15b0ae057c8d/src/groups/bmq/bmqp/bmqp_ctrlmsg.xsd#L382-L398):
+
+- `appId`: This string contains the name of *appId*, if any, for the consumer.
+  *AppId*s are specified by consumers who are attaching to a *fan-out* queue.
+  Consumers of non-fan-out queues, as well as any producers don't need to
+  specify any appId.
+
+- `subscriptions`: This field is an array of zero or more
+  [`Subscription`](https://github.com/bloomberg/blazingmq/blob/9b692fe25f74543e954a27e30b6b15b0ae057c8d/src/groups/bmq/bmqp/bmqp_ctrlmsg.xsd#L455-L471)s.
+  Each `Subscription` type contains these fields:
+
+  - `sId`: This integer represents a unique identifier for this subscription.
+    This integer needs to be unique only between this client and the broker.  A
+    simple incrementing
+    [variable](https://github.com/bloomberg/blazingmq/blob/9b692fe25f74543e954a27e30b6b15b0ae057c8d/src/groups/bmq/bmqt/bmqt_subscription.cpp#L44-L49)
+    can be used for generating unique `sId`s.
+
+  - `expression`: This field, which is of type
+    [`Expression`](https://github.com/bloomberg/blazingmq/blob/9b692fe25f74543e954a27e30b6b15b0ae057c8d/src/groups/bmq/bmqp/bmqp_ctrlmsg.xsd#L412C4-L426)
+    contains the actual subscription expression as a string type as well as
+    expression grammar version.
```

Comment:
```suggestion
    the version of the expression grammar.
```
Was confused at first by this phrasing: didn't know what it meant for the type to contain both a string version of the subscription expression and an expression grammar version of the subscription expression.  This edit makes it clear that it's a version identifier as per the schema.